core/contract/src/main/java/org/eclipse/dataspaceconnector/contract/ContractServiceExtension.java

```diff
@@ -162,6 +167,8 @@ private void registerServices(ServiceExtensionContext context) {
         CommandRunner<ContractNegotiationCommand> commandRunner = new CommandRunner<>(commandHandlerRegistry, monitor);
 
         var observable = new ContractNegotiationObservableImpl();
+        observable.registerListener(new ContractNegotiationEventListener(eventRouter, clock));
+
         context.registerService(ContractNegotiationObservable.class, observable);
```

Comment:
not directly related, but maybe use `@Provider` methods?

Reply:
@paullatzelsperger as I specified in the last PR, the `@Provider` method does not work in certain conditions (at a first sight seems like you can't use injected services into a `@Provider` method, but I didn't dive into it):
https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/pull/1587#discussion_r913881102

Reply:
I guess the problem you're alluding to has to do with the sequence in which the phases of the DI process are executed. However, I managed to get this particular example to work (see [gist](https://gist.github.com/paullatzelsperger/5538411d2b269251c4acd8f0b7272c7d)).

Reply:
@paullatzelsperger ok, but the gist code moves to `@Provider` methods only stuff that's not related to the PR, I would address this change to another PR.